src/components/TimeseriesSearch/stories/basic.story.mdx

```diff
@@ -0,0 +1,224 @@
+  list?: React.CSSProperties;
+  selectAllButton?: React.CSSProperties;
+  selectNoneButton?: React.CSSProperties;
+}
+```
+
+See more details in `Custom styles` example.
+
+## Examples
+
+### Show Root Asset Select
+
+#### Info: 
+
+<p>For all examples You can search for {names}</p>
+
+<Preview>
+  <Story name="Show Root Asset Select" >
+    <TimeseriesSearch
+      onTimeserieSelectionChange={onTimeserieSelectionChange}
+      rootAssetSelect={true}
+    />
+  </Story>
+</Preview>
+
+### Hide Selected Row
+
+<Preview>
+  <Story name="Hide Selected Row" >
+    <TimeseriesSearch
+      onTimeserieSelectionChange={onTimeserieSelectionChange}
+      hideSelected={true}
+    />
+  </Story>
+</Preview>
+
+### Single Selection
+
+<Preview>
+  <Story name="Single Selection" >
+    <TimeseriesSearch
+      onTimeserieSelectionChange={onTimeserieSelectionChange}
+      single={true}
+    />
+  </Story>
+</Preview>
+
+### Allow Strings
+
+<Preview>
+  <Story name="Allow Strings" >
+    <TimeseriesSearch
+      onTimeserieSelectionChange={onTimeserieSelectionChange}
+      allowStrings={true}
+    />
+  </Story>
+</Preview>
+
+### Preselected
+
+<Preview>
+  <Story name="Preselected" >
+    <TimeseriesSearch
+      onTimeserieSelectionChange={onTimeserieSelectionChange}
+      selectedTimeseries={[8681821313339919, 285034733763395]}
+    />
+  </Story>
+</Preview>
+
+### Custom Filter Rule
+
+```jsx
+const filterRule = (timeseries: GetTimeSeriesMetadataDTO): boolean =>
+  !timeseries.isString;
+```
+
+<Preview>
+  <Story name="Custom Filter Rule" >
+    <TimeseriesSearch
+      onTimeserieSelectionChange={onTimeserieSelectionChange}
+      filterRule={filterRule}
+    />  
+  </Story>
+</Preview>
+
+### Custom Strings
+
+```jsx
+const customString = {
+  rootAssetSelectAll: 'No filter',
+  searchPlaceholder: 'search for stuff!',
+  selectAll: 'EfilterRuleverything!',
+  selectNone: 'Nothing!',
+};
+
+```
+
+<Preview>
+  <Story name="Custom Strings" >
+    <TimeseriesSearch
+      onTimeserieSelectionChange={onTimeserieSelectionChange}
+      rootAssetSelect={true}
+      strings={customString}
+    />  
+  </Story>
+</Preview>
+
+### With Custom Styles
+
+```jsx
+const customStyles = {
+  list: { height: '200px' },
+  buttonRow: { marginTop: '30px' },
+  selectAllButton: { backgroundColor: 'lightblue' },
+  selectNoneButton: {
+    backgroundColor: 'magenta',
+    marginLeft: '50px',
+  },
+};
+```
+
+<Preview>
+  <Story name="With Custom Styles" >
+    <TimeseriesSearch
+      onTimeserieSelectionChange={onTimeserieSelectionChange}
+      styles={customStyles}
+    />  
+  </Story>
+</Preview>
+
+### With Theme
+
+```jsx
+import { ThemeProvider } from 'styled-components';
+
+const ExampleTheme = {
+  gearbox: {
+    selectColor: 'red',
+    white: '#fff',
+  },
+};
+```
+
+<Preview>
+  <Story name="With Theme" >
+    <ThemeProvider theme={ExampleTheme}><TimeseriesSearch onTimeserieSelectionChange={onTimeserieSelectionChange}/></ThemeProvider>
```

Comment:
alignment

Reply:
This should be kept inline for showing better code snippets in docs.

Reply:
got it